### PR TITLE
document process application milestone versioning

### PR DESCRIPTION
**Pull request by christinaausley**
## Description

Related to https://github.com/camunda/web-modeler/issues/9808.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/components/modeler/web-modeler/milestones.md [hunk 1]
```diff
@@ -25,6 +25,13 @@ Alternatively, you can create a new milestone via the breadcrumb menu in the dia
 
 When dragging and dropping a file into the diagram view, or when using the **Replace via upload** option under the breadcrumb menu, a new milestone is created automatically.
 
+#### Bulk milestone creation
+
+[Process applications](/components/modeler/web-modeler/process-applications.md) are a special type of folder in Web Modeler that allows you to work on a set of related files and
+[deploy](/components/modeler/web-modeler/process-applications.md#deploy-and-run-a-process-application) them together in a single bundle with just one click. This reduces the risk of having a broken deployment at runtime and makes it more convenient to deploy related files.
+
+When creating a milestone on a main process of a process application, milestones are created for all other assets in the application to make it easier to track or roll back changes. Depending on your solutions approach, these milestones are either independently managed from the main process, or are tied to the main process and cannot be modified.
```
Comment (vpellegrino) [inline docs/components/modeler/web-modeler/milestones.md:33]: For the solution we just implemented, the milestones of the resources belonging to a process application are tied to the main process and cannot be modified.

File: docs/components/modeler/web-modeler/process-applications.md [hunk 2]
```diff
@@ -106,6 +106,16 @@ Note that when you deploy the process application:
 - Linked external forms will be deployed together with the process application.
 - Linked external BPMN and DMN diagrams are _not_ deployed together. They must be deployed separately.
 
+## Milestones
```
Comment (vpellegrino) [inline docs/components/modeler/web-modeler/process-applications.md:109]: In the context of a process application, we should only refer to `Versioning`, and avoid mentioning "Milestones", even if, from the implementation perspective, we are still using milestones behind the scenes.
So, I suggest not mentioning milestones at all on this doc page.
Involving here also @HanselIdes and @RomanKostka.

File: docs/components/modeler/web-modeler/process-applications.md [hunk 3]
```diff
@@ -106,6 +106,16 @@ Note that when you deploy the process application:
...
+
+You can save a snapshot of your diagram as a milestone any time. If you make any mistakes while modeling, you can always go back to previously saved snapshots. You can also identify and compare the differences between two milestones.
+
+You can also add version tags to these milestones, and create milestones for all resources of given process applications. For more information, review the [milestone documentation](/components/modeler/web-modeler/milestones.md).
+
+### Versioning
+
+When creating a milestone, name your milestone with a version tag to track it through the development lifecycle and ensure the correct version is called. To do this, enter a **Version tag** within the **Create a milestone** modal and click **Create**.
+
 ## Limitations
```
Comment (vpellegrino) [inline docs/components/modeler/web-modeler/process-applications.md:119]: In the "Limitations" section, should we also mention that modifying/deleting the created versions is impossible? @RomanKostka

Comment (RomanKostka) [reply docs/components/modeler/web-modeler/process-applications.md:119]: Yes, that is an important limitation and we should mention it

File: docs/components/modeler/web-modeler/milestones.md (hunk 1 above)
Comment (christinaausley) [reply docs/components/modeler/web-modeler/milestones.md:33]: Adjusted in latest commit 👍

File: docs/components/modeler/web-modeler/process-applications.md (hunk 2 above)
Comment (christinaausley) [reply docs/components/modeler/web-modeler/process-applications.md:109]: Adjusted in my latest commit, but let me know what you think.

File: docs/components/modeler/web-modeler/process-applications.md (hunk 3 above)
Comment (christinaausley) [reply docs/components/modeler/web-modeler/process-applications.md:119]: Added in latest commit 👍

**Review by vpellegrino — APPROVED** [review]
✅ Great job, @christinaausley!
I like the way the information is simple and accessible.
I am approving it from my side.
Consider that @HanselIdes is on FTO for a few weeks, so feel free to merge this doc change if you want.

**Review by RomanKostka — APPROVED** [review]
looks good from my end. Thanks!

**Review by HanselIdes — COMMENTED** [review]
LGTM (with one grammatical issue)

File: docs/components/modeler/web-modeler/milestones.md [hunk 4]
```diff
@@ -25,6 +25,17 @@ Alternatively, you can create a new milestone via the breadcrumb menu in the dia
 
 When dragging and dropping a file into the diagram view, or when using the **Replace via upload** option under the breadcrumb menu, a new milestone is created automatically.
 
+#### Bulk milestone creation
+
+[Process applications](/components/modeler/web-modeler/process-applications.md) are a special type of folder in Web Modeler that allows you to work on a set of related files and
```
[the remark this answers is not in the document]
Comment (christinaausley) [reply docs/components/modeler/web-modeler/milestones.md:30]: I'll adjust this in another PR 👍 Thanks for the catch.

**Pull request merged**